File: sic-api/Entities/Su/SuUserBusiness.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace sic_api.Entities.Su;

[Table("su_user_business")]
public class SuUserBusiness : BaseEntity
{
    [Required]
    [MaxLength(100)]
    [Column("user_id")]
    public string UserId { get; set; } = default!;

    [Required]
    [ForeignKey(nameof(Business))]
    [Column("business_id")]
    public Guid BusinessId { get; set; }
    public SuBusiness Business { get; set; } = default!;

    [Required]
    [Column("is_default")]
    public bool IsDefault { get; set; } = false;

    [Required]
    [Column("is_active")]
    public bool IsActive { get; set; } = false;
    
    public ICollection<SuUserBusinessRole>? UserBusinessRoles { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuUserBusiness.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "su_user_business",
       indexes = {
           @Index(name = "idx_user_business", columnList = "user_id, business_id", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuUserBusiness extends BaseEntity {

    @Column(name = "user_id", nullable = false, length = 100)
    private String userId;

    @Column(name = "business_id", insertable = false, updatable = false)
    private UUID businessId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "business_id", nullable = false)
    private SuBusiness business;

    @Column(name = "is_default", nullable = false)
    private Boolean isDefault = false;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

    @OneToMany(mappedBy = "userBusiness", fetch = FetchType.LAZY)
    private List<SuUserBusinessRole> userBusinessRoles = new ArrayList<>();
}